### DISCUSSION TITLE
burst2stack.exe not exist.

### DISCUSSION THREAD

**Discussion by AhmedAthab** [ASFHyP3/burst2safe#95] [Q&A] [answered]
Hi, I installed the latest version of burst2safe using conda install -c conda-forge burst2safe. I can see the burst2safe.exe file in the environment folder, but the burst2stack.exe is not available. there is only one file called burst2stack that has no extension. This file was available in the previous version. any idea why it is missing?

**Comment by joaocarr** [comment]
Hi, I can confirm the same happening when using "conda install -c conda-forge burst2safe". I cannot run burst2stack with my installation of  burst2safe v1.4.2.

**Comment by joaocarr** [comment]
Don't want to be rude, but I wonder if any dev people are really looking at the issues and discussions pages

**Comment by jtherrmann** [comment]
@AhmedAthab @joaocarr I cannot reproduce this issue on my Linux machine (Ubuntu 24.04.2 LTS with conda 25.1.1). I ran the following commands:

```shell
conda create -n test-burst2safe python=3.12
conda activate test-burst2safe
conda install -c conda-forge burst2safe
burst2stack -h
burst2stack --rel-orbit 64 --start-date 2020-06-03 --end-date 2020-06-17 --extent 53.57 27.54 53.78 27.60
```

And the final command completed successfully with the following output:

```
Found 4 burst(s), comprising 2 SAFE(s).
Check burst group validities...
Downloading data...
Download complete.
Creating SAFEs...
SAFEs creaated!
```

From the date that this discussion was originally opened (Nov 9, 2024), I assume that the problem may have started when @AhmedAthab installed [burst2safe v1.3.0](https://github.com/ASFHyP3/burst2safe/releases/tag/v1.3.0) (released on Nov 6, 2024). However, I do not see anything in the release notes or the [diff](https://github.com/ASFHyP3/burst2safe/pull/94/files) for that release that would have affected `burst2stack`, as I'm only seeing changes to `local2safe`.

Can you please provide as much information as possible about your system? Including:

- What is your operating system and version?
- If on Windows or macOS, please provide as much information as possible about your command-line environment, including terminal emulator and shell, as I am not very familiar with those operating systems.
- What is your conda version (run `conda --version`)? How was it installed? If you can link to the installation steps you followed, that would be ideal.
- What Python version is installed in your conda environment (activate your conda environment and run `python --version`)?
- Please verify you have the latest version of burst2safe installed (v1.4.2)
- What are the exact commands that you're using to create your conda environment, install burst2safe, and run `burst2stack`? What error message(s) or output are you seeing (or not seeing) that indicate there's a problem?

Please note that this is an open-source project and while we do our best to answer user support questions, we do not necessarily have the resources to support every configuration or use-case. We'll do our best to help, but please be patient as we have a lot on our plate.

**Reply by joaocarr** [reply]
Dear Jake,
apologies if my comment came across as rude, it wasn't my intention. I genuinely was checking if this page was being monitored. Your efforts in providing open source solutions to deal with S-1 burst data is highly appreciated. 

I am still facing some issues with having a running installation of burst2safe. Let me provide you with my details:

1. Windows 10 on a Win-64 machine

2. I have Miniconda installed and running my commands on an Anaconda prompt

3. conda 24.9.2, I did a clean install using:
conda create --prefix C:\Dev\conda_envs\b2safe
conda activate C:\Dev\conda_envs\b2safe
conda install -c conda-forge burst2safe

4. Python 3.12.9

5. burst2safe, 1.4.2, pyhd8ed1ab_0, conda-forge

6. burst2safe -h

Traceback (most recent call last):
  File "C:\Dev\conda_envs\b2safe\Lib\site-packages\osgeo\__init__.py", line 30, in swig_import_helper
    return importlib.import_module(mname)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Dev\conda_envs\b2safe\Lib\importlib\__init__.py", line 90, in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "<frozen importlib._bootstrap>", line 1387, in _gcd_import
  File "<frozen importlib._bootstrap>", line 1360, in _find_and_load
  File "<frozen importlib._bootstrap>", line 1331, in _find_and_load_unlocked
  File "<frozen importlib._bootstrap>", line 921, in _load_unlocked
  File "<frozen importlib._bootstrap>", line 813, in module_from_spec
  File "<frozen importlib._bootstrap_external>", line 1293, in create_module
  File "<frozen importlib._bootstrap>", line 488, in _call_with_frames_removed
ImportError: DLL load failed while importing _gdal: The specified procedure could not be found.

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "C:\Dev\conda_envs\b2safe\Scripts\burst2safe-script.py", line 5, in <module>
    from burst2safe.burst2safe import main
  File "C:\Dev\conda_envs\b2safe\Lib\site-packages\burst2safe\burst2safe.py", line 10, in <module>
    from burst2safe import utils
  File "C:\Dev\conda_envs\b2safe\Lib\site-packages\burst2safe\utils.py", line 14, in <module>
    from osgeo import gdal, ogr, osr
  File "C:\Dev\conda_envs\b2safe\Lib\site-packages\osgeo\__init__.py", line 35, in <module>
    _gdal = swig_import_helper()
            ^^^^^^^^^^^^^^^^^^^^
  File "C:\Dev\conda_envs\b2safe\Lib\site-packages\osgeo\__init__.py", line 32, in swig_import_helper
    return importlib.import_module('_gdal')
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Dev\conda_envs\b2safe\Lib\importlib\__init__.py", line 90, in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
ModuleNotFoundError: No module named '_gdal'

**Reply by forrestfwilliams** [reply]
@joaocarr It looks like gdal is not being installed correctly on windows for you.
Try modifying the install commands in step 3 to:

```
conda create --prefix C:\Dev\conda_envs\b2safe
conda activate C:\Dev\conda_envs\b2safe
conda install -c conda-forge gdal
conda install -c conda-forge burst2safe
```

**Reply by joaocarr** [reply]
@forrestfwilliams thanks for reaching out, unfortunately that didn't work, still getting the same error

**Reply by joaocarr** [reply]
@forrestfwilliams  Quick update about burst2safe in Linux: I have it running with no problems whatsoever, the issue is with some configuration in Windows. I just need it at this point to be running with Windows because of a specific dev version of SNAP I'm using that I cannot have installed in my Linux environment. So, my workaround at the moment is to be moving burst SAFE files from linux to windows.

**Reply by jhkennedy** [reply, marked as answer]
@AhmedAthab and @joaocarr the issue is with the conda-forge feedstock. 

When the burst2stack entrypoint was added to the project, it was not reflected in the conda-forge recipe. This PR should fix the issue once merged:
https://github.com/conda-forge/burst2safe-feedstock/pull/31

**Reply by jhkennedy** [reply]
@AhmedAthab and @joaocarr,  the new conda package build is available for install. Can you try it out and let us know if it solves your issue?

You should be able to upgrade to the latest build with a `conda install burst2safe`, but if that doesn't work, you can specify the new build directly with `conda install burst2safe=1.4.2=pyhd8ed1ab_1`.

**Reply by joaocarr** [reply]
@jhkennedy , now running burst2stack -h does not give me an error saying it doesn't exist. But I'm still getting that gdal error, which is unrelated to your package. And this only happens when I'm running it using Windows. Thank you ever so much for keeping this package up to date.

**Reply by jhkennedy** [reply]
@joaocarr glad to hear burst2stack is no longer giving you a  does not exist error! I'm going to mark this discussion as answered then.

For the GDAL issue, I'm not sure we'll be able to help much, and you'll likely have better luck by opening an issue here: 
https://github.com/conda-forge/gdal-feedstock

That said, a little searching suggests installing `krb5` might help:
```
conda install -c conda-forge krb5
```
https://github.com/OSGeo/gdal/issues/2957#issuecomment-1669150443

**Reply by jhkennedy** [reply]
@joaocarr @AhmedAthab thank you for notifying us and for your patience!
